### ISSUE TITLE
Failing to scale out node pool

### ISSUE THREAD

**Issue by oryagel** [Azure/acs-engine#1412]
**Is this a request for help?**
Yes

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one)
Issue

---

**What version of acs-engine?**
0.5.0

---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes 1.7.2

**What happened**:
I tried to add nodes to an existing cluster. Followed this guide: https://github.com/Azure/acs-engine/blob/master/examples/scale-up/README.md
I get many errors when deploying: `Changing property 'customData' is not allowed`

**What you expected to happen**:
The deployment should succeed and add more nodes.

**Comment by ITler** [comment]
Applies to me, too, but in version 0.6.0 with Kubernetes `1.6.9` on *AzureGermanCloud*.

1. I use `acs-engine generate k8s.json` to generate `azuredeploy.json` and `azuredeploy.parameters.json`, which can be deployed to Azure successfully. Fine so far.
2. Based on https://github.com/Azure/acs-engine/blob/master/examples/scale-up/README.md I try to scale up and therefore remove every resource from JSON file, which is depending on `nsgID`.

However this results in removing i.e. NIC definitions (`"type": "Microsoft.Network/networkInterfaces"`) as they depend on the `networkSecurityGroup` identified by `nsgID`. This is senseless, as this would lead to VMs without NIC?!

We are about to decide, if we use Microsoft Azure as a service provider and sign contract or not to host hundreds of our customers. As we need to use _AzureGermanCloud_ for legal reasons, we only have the chance to use `acs-engine` and use *Azure Deployment Templates* to bootstrap and scale K8s with an automated approach. Using *AzureCli* with commands like `az acs [...]` is not possible, because *AzureGermanCloud* does not implement providers like `Microsoft.ContainerServices`. So being not able to scale *by code* currently is a blocker for business.

**Comment by markya0616** [comment]
Does anyone have idea about this issue? In our case, we encountered the similar issue. We use `acs-engine generate` the new `parameter.json` to add more agent pools. Then use `az group deployment create` to update the same cluster, but got the same error.

```
{
  "error": {
    "code": "PropertyChangeNotAllowed",
    "target": "customData",
    "message": "Changing property 'customData' is not allowed."
 }
```

However, we checked the azure portal, and the new agent pools are created. Not sure whether it can work correctly. The Kubernetes version is v1.7.5.

**Comment by endeepak** [comment]
Same issue with docker swarm setup after changing the `postInstallScriptURI` parameter.  Seems to be similar to https://github.com/Azure/acs-engine/issues/132

**Comment by CecileRobertMichon** [comment]
The scale command should be used to scale up/down. Please see https://github.com/Azure/acs-engine/blob/master/docs/kubernetes/scale.md for detailed instructions.

**Comment by ITler** [comment]
... which wasn't available in 0.5.0

**Comment by birdemily** [comment]
@CecileRobertMichon can't scale windows node, "index error". Still have this issue.
